config/src/network_id.rs

```diff
@@ -13,6 +13,10 @@ pub struct NetworkContext {
     network_id: NetworkId,
     role: RoleType,
     peer_id: PeerId,
+
+    /// Cache rendering the short PeerId String
+    #[serde(skip)]
+    peer_id_short_str: String,
```

Comment:
Personally, this always gets confusing to me when it's named `str` but it's a `String`

Reply:
for sure, i can change the field to `peer_id_short_string` and keep the acessor as `peer_id_short_str() -> &str`

Reply:
Oh, I wouldn't rename it.  I was just pointing it out, because `peer_id.short_str()` exists.

Reply:
Maybe you can avoid the Alloc altogether and create a wrapping type `struct ShortPeerId<'a>(&'a PeerId)` which just has a display impl that does the short printing

Reply:
I actually wrote something like that a long time ago but forgot to merge it in... I'll open a follow up PR with it

Reply:
What motivated this change? This feels like a premature optimization.